### ISSUE TITLE
[BUG] LCSS and EDR PAM doesnt assign correct amount of clusters

### ISSUE THREAD

**Issue by chrisholder** [aeon-toolkit/aeon#882]
### Describe the bug

For PAM-LCSS and PAM-EDR sometimes the correct amount of clusters are not assigned. Specifically LCSS on Adiac and EDR on OliveOil assign incorrect number of clusters.

### Expected results

Should have clusters = n_clusters

### Actual results

cluster != n_clusters

**Comment by TonyBagnall** [comment]
I can reproduce this like so

```python
from aeon.clustering import TimeSeriesKMedoids
from aeon.datasets import load_classification
from aeon.distances import lcss_distance, edr_distance

trainX, trainy = load_classification(name="OliveOil",split="train")
testX, testy = load_classification(name="OliveOil",split="test")

cls = np.unique(trainy)
cls2 = np.unique(testy)

print(" Train classes = ",cls, " length ",len(cls))
print(" Train classes = ",cls2, " length ",len(cls2))

knn = TimeSeriesKMedoids(distance=lcss_distance, n_clusters=len(cls), method="pam")
knn.fit(trainX)
preds = knn.predict(trainX)
preds2 = knn.predict(testX)
print(" Predicted classes = ",np.unique(preds), " Test classes = ",np.unique(preds2))


trainX, trainy = load_classification(name="Adiac",split="train")
testX, testy = load_classification(name="Adiac",split="test")

cls = np.unique(trainy)
cls2 = np.unique(testy)

print(" Train classes = ",cls, " length ",len(cls))
print(" Train classes = ",cls2, " length ",len(cls2))

knn = TimeSeriesKMedoids(distance=lcss_distance, n_clusters=len(cls), method="pam")
knn.fit(trainX)
preds = knn.predict(trainX)
preds2 = knn.predict(testX)
print(" Predicted classes = ",np.unique(preds), " Test classes = ",np.unique(preds2))
```
output for olive oil 
```python-traceback
Train classes =  ['1' '2' '3' '4']  length  4
 Train classes =  ['1' '2' '3' '4']  length  4
 Predicted classes =  [0]  Test classes =  [0]
```

**Comment by chrisholder** [comment]
I think this is now fixed I get the output:
 Predicted classes =  [ 0  1  7  8 10]  Test classes =  [ 0  1  3  9 10 17]

**Comment by dguijo** [comment]
I'm facing some issues with these two distances too. 
In this case, I'm playing around kNN versions of aeon and sklearn.

I attach the source code in a file ([test_LCSS_EDR.py.zip](https://github.com/user-attachments/files/16400900/test_LCSS_EDR.py.zip)). It tests 784 combinations of kNNs comparisons:
- 2 types of tasks: classification and regression.
- 2 types of datasets: univariate and multivariate.
- 16 distances.
- 3 values for the number of neighbours.

For all of these 196 runs, we have 4 tests:
- Loading univariate problem as 2D, which is compatible with sklearn's knn.
- Using sklearn's knn with precomputed distances.
- Loading multivariate problem as 3D, but concatenating channels to transform it to 2D, to be compatible with sklearn's knn. 
- Using sklearn's knn with precomputed distances (keeping the channels). 

From these 784 combinations, LCSS and EDR do not achieve the same results using aeon's and sklearn's kNN versions in 18 times. LCSS fails 8 times and EDR 10 times. 

Thank you!

**Comment by TonyBagnall** [comment]
@dguijo this original issue is fixed, cant reproduce the original. Could you create a new issue re KNN if your bug is still relevant, and include a small example, tagged as distances and classification?

**Comment by TonyBagnall** [comment]
premature closure!

**Comment by chrisholder** [comment]
This is a threshold issue with edr and lcss. If you set a proper threshold it returns differences.

**Comment by TonyBagnall** [comment]
```python
import numpy as np
from aeon.clustering import TimeSeriesKMedoids
from aeon.datasets import load_classification
from aeon.distances import lcss_distance, edr_distance, dtw_distance
from aeon.transformations.collection import Normalizer
trainX, trainy = load_classification(name="OliveOil",split="train")
testX, testy = load_classification(name="OliveOil",split="test")

cls = np.unique(trainy)
cls2 = np.unique(testy)


print(" Train classes = ",cls, " length ",len(cls))
print(" Train classes = ",cls2, " length ",len(cls2))
dist_params = {"epsilon":0.00000000}
knn = TimeSeriesKMedoids(distance=lcss_distance, n_clusters=len(cls),
                         method="pam", distance_params=dist_params)
knn.fit(trainX)
preds = knn.predict(trainX)
preds2 = knn.predict(testX)
print(" Predicted classes = ",np.unique(preds), " Test classes = ",np.unique(preds2))
print(knn.labels_)
``
